packages/record-data/addon/-private/graph/index.ts

```diff
@@ -26,7 +27,17 @@ import updateRelationshipOperation from './operations/update-relationship';
 
 type RelationshipEdge = ImplicitRelationship | ManyRelationship | BelongsToRelationship;
 
-const Graphs = new WeakMap<RecordDataStoreWrapper, Graph>();
+const Graphs = new WeakCache<RecordDataStoreWrapper, Graph>(DEBUG ? 'graph' : '');
+Graphs._generator = (wrapper: RecordDataStoreWrapper) => {
+  const graph = new Graph(wrapper);
+
+  // in DEBUG we attach the graph to the main store for improved debuggability
+  if (DEBUG) {
+    Graphs.set(wrapper._store as unknown as RecordDataStoreWrapper, graph);
```

Comment:
Doesn't this happen when `lookup` is called?  Or is this adding the graph to an additional object?

Reply:
@snewcomer we're adding it to an additional object here. The cache is key'd to the store-wrapper not the store in the map, but for debugging we also attach it to the store.